File: frontend/src/types/community.ts
export type EntryCategory =
  | "attraction"
  | "hotel"
  | "guesthouse"
  | "homestay"
  | "restaurant";

export type EntryTag = "trending" | "hidden_gem";

export type PriceRange = "budget" | "mid_range" | "premium" | "luxury";

export interface PhotoItem {
  id: string;
  url: string;
  public_id: string;
  caption: string | null;
}

export interface VideoEmbed {
  id: string;
  url: string;
  platform: "youtube" | "facebook" | "tiktok";
}

export interface CommunityEntry {
  id: string;
  user_id: string;
  author_name: string;
  author_picture_url: string | null;
  category: EntryCategory;
  name: string;
  location: string;
  latitude: number | null;
  longitude: number | null;
  price_range: PriceRange;
  amenities: string[];
  photos: PhotoItem[];
  travel_tips: string;
  video_embeds: VideoEmbed[];
  tags: EntryTag[];
  created_at: string;
  updated_at: string;
}

export interface CommunityEntryListParams {
  page?: number;
  per_page?: number;
  category?: EntryCategory;
  tag?: EntryTag;
  search?: string;
  sort_by?: "newest" | "name";
}

export interface PaginatedResponse<T> {
  items: T[];
  total: number;
  page: number;
  per_page: number;
  total_pages: number;
}

export type CreateEntryPayload = {
  category: EntryCategory;
  name: string;
  location: string;
  latitude?: number | null;
  longitude?: number | null;
  price_range: PriceRange;
  amenities: string[];
  travel_tips: string;
  video_embeds: { url: string; platform: "youtube" | "facebook" | "tiktok" }[];
  tags: EntryTag[];
};

export type UpdateEntryPayload = Partial<CreateEntryPayload>;

File: backend/app/schemas/community.py
import uuid
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, ConfigDict, HttpUrl


class VideoEmbedBase(BaseModel):
    url: str
    platform: str # youtube, facebook, tiktok


class VideoEmbedCreate(VideoEmbedBase):
    pass


class VideoEmbedRead(VideoEmbedBase):
    id: uuid.UUID

    model_config = ConfigDict(from_attributes=True)


class PhotoRead(BaseModel):
    id: uuid.UUID
    url: str
    public_id: str
    caption: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class CommunityEntryBase(BaseModel):
    category: str
    name: str
    location: str
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    price_range: str
    amenities: List[str] = []
    travel_tips: Optional[str] = None
    tags: List[str] = []


class CommunityEntryCreate(CommunityEntryBase):
    video_embeds: List[VideoEmbedCreate] = []


class CommunityEntryUpdate(BaseModel):
    category: Optional[str] = None
    name: Optional[str] = None
    location: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    price_range: Optional[str] = None
    amenities: Optional[List[str]] = None
    travel_tips: Optional[str] = None
    tags: Optional[List[str]] = None
    video_embeds: Optional[List[VideoEmbedCreate]] = None


class CommunityEntryRead(CommunityEntryBase):
    id: uuid.UUID
    user_id: uuid.UUID
    created_at: datetime
    updated_at: datetime
    
    photos: List[PhotoRead] = []
    video_embeds: List[VideoEmbedRead] = []
    
    # These will be populated by the service layer via join
    author_name: str
    author_picture_url: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class CommunityMapPoint(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    name: str
    location: str
    latitude: float
    longitude: float
    category: str
    tags: List[str] = []
    author_name: str
    author_picture_url: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class PaginatedResponse(BaseModel):
    total: int
    page: int
    per_page: int
    total_pages: int


class CommunityEntryList(PaginatedResponse):
    items: List[CommunityEntryRead]
